spec/unit_tests/moab/storage_object_spec.rb

```diff
@@ -1,551 +1,701 @@
 # frozen_string_literal: true
 
 describe Moab::StorageObject do
+  let(:ingests_object_dir) { ingests_dir.join(BARE_TEST_DRUID) }
+  let(:storage_object_from_ingests) { described_class.new(FULL_TEST_DRUID, ingests_object_dir) }
+  let(:temp_dir_ingests) { temp_dir.join('ingests') }
+  let(:temp_object_dir) { temp_dir_ingests.join(BARE_TEST_DRUID) }
+  let(:storage_object_from_temp) do
+    storage_object_from_temp = described_class.new(FULL_TEST_DRUID, temp_object_dir)
+    storage_object_from_temp.initialize_storage
+    storage_object_from_temp.storage_root = fixtures_dir.join('derivatives')
+    storage_object_from_temp
+  end
   let(:eq_xml_opts) { { element_order: false, normalize_whitespace: true } }
-  let(:storage_object) { described_class.new(FULL_TEST_DRUID, @ingest_object_dir) }
 
   before do
-    @storage_object = described_class.new(FULL_TEST_DRUID, @temp_object_dir)
-    @storage_object.initialize_storage
-    @storage_object.storage_root = fixtures_dir.join('derivatives')
+    temp_object_dir.rmtree if temp_object_dir.exist?
   end
 
-  after(:all) do
-    @temp_dir_ingests.rmtree if !@temp_dir_ingests.nil? && @temp_dir_ingests.exist?
+  after do
+    temp_dir_ingests.rmtree if temp_dir_ingests&.exist?
```

Comment:
Wondering why `temp_dir_ingests` might be nil?

Reply:
valid question ...